src/deploy/001_authenticator.ts

```diff
@@ -0,0 +1,25 @@
+import { DeployFunction } from "hardhat-deploy/types";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+import { salt, logResult, contractNames } from "../ts/deploy";
+
+const deployAuthenticator: DeployFunction = async function (
+  hre: HardhatRuntimeEnvironment,
```

Comment:
ubernit: You can decontruct the env already in the function signature:
```suggestion
  { deployments, getNamedAccounts, network }: HardhatRuntimeEnvironment,
```

Reply:
Very much like the suggestion. Otherwise, can we call it `runtime` or `environment` instead of a cryptic abbreviation?

Reply:
One thing to note is that hardhat documentation uses `hre`. So in cases where we don't use/can't use destructuring I would still prefer `hre` as it will be easier to find documentation and examples using that cryptic abbreviation (even if I'm not a fan of it).

Reply:
Object is now destructured in the function argument, it looks nicer. Indeed I went for `hre` to be consistent with the docs.